### PR TITLE
Add guide for writing e2e tests in README

### PR DESCRIPTION
**Pull request by juliaamosova**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR adds a detailed guide on how to write e2e tests in README. The guide contains the following sections:

Guide for writing e2e tests
  - Tools for writing tests
  - Creating test structure
  - Writing the test
  - Best practices

### How to test the changes in this Pull Request:

This PR adds documentation only, therefore, no need to actually test it but rather review to make sure it covers all the necessary details for writing e2e tests. 

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### Changelog entry

N/A since we don't include e2e tests work in the releases.

### CODE REVIEW

**Review by rrennick — DISMISSED** [review]
@juliaamosova Looks good.

**Comment by juliaamosova** [comment]
Thanks for taking a look so quickly @rrennick! I want to add a few more things to this PR which is why I labeled it as `in progress` :)

**Comment by juliaamosova** [comment]
@rrennick @tammullen this PR is ready to be reviewed again.

File: tests/e2e/README.md [hunk 1]
```diff
@@ -226,20 +230,102 @@ The full command to build the site will look as follows:
...
+### Tools for writing tests
 
 We use the following tools to write e2e tests:
 
 - [Puppeteer](https://github.com/GoogleChrome/puppeteer) – a Node library which provides a high-level API to control Chrome or Chromium over the DevTools Protocol
 - [jest-puppeteer](https://github.com/smooth-code/jest-puppeteer) – provides all required configuration to run tests using Puppeteer
 - [expect-puppeteer](https://github.com/smooth-code/jest-puppeteer/tree/master/packages/expect-puppeteer) – assertion library for Puppeteer
 
-Tests are kept in `tests/e2e/specs` folder.
+Tests are kept in `tests/e2e/core-tests/specs/` folder.
```
Comment (rrennick) [inline tests/e2e/README.md:243]: Adding WC core tests should follow the instructions in #27916. In other repos using our e2e packages the tests should be in `tests/e2e/specs/`.

**Review by rrennick — DISMISSED** [review]
@juliaamosova This looks great other than the one clarification I noted. It should be okay to temporarily have a 404 on `tests/e2e/core-tests/README.md` between when the two PRs are merged. Pre-approving with needing that one change.

File: tests/e2e/README.md (hunk 1 above)
Comment (juliaamosova) [reply tests/e2e/README.md]: Thanks for this @rrennick! Corrected. Ready for another review.

**Review by rrennick — APPROVED** [review]
@juliaamosova Looks great.

**Pull request merged**